Solutions/Marain.Workflow.Abstractions/Marain/Workflows/HostedWorkflowTriggerParameterTestType.cs

```diff
@@ -0,0 +1,32 @@
+﻿// <copyright file="HostedWorkflowTriggerParameterTestType.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Workflows
+{
+    /// <summary>
+    /// Types of test for the <see cref="HostedWorkflowTriggerParameterCondition"/>.
+    /// </summary>
+    public enum HostedWorkflowTriggerParameterTestType
+    {
+        /// <summary>
+        /// The parameter value should match the value specified in the condition.
+        /// </summary>
+        Equality,
```

Comment:
Have you considered Contains, DoesNotContain, LessThan, GreaterThan, LessThan, GreaterThanOrEquals, LessThanOrEquals?

Reply:
Yes, but I'd like to park these for now as I think the ongoing effort to use `IPropertyBag` instead of `IDictionary<string, string>` will have an impact on their implementation. For example, with GreaterThan/LessThan, once we have moved to `IPropertyBag` then we will have to determine what type to treat the parameter as when doing the check.

It should be possible to add further tests without it being a breaking change.